### PR TITLE
Split Solana fees

### PR DESCRIPTION
**Pull request by 0xBoxer**
This PR splits Solana fees into separate components.

### CODE REVIEW

File: dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_fees.sql
Comment (0xRobin) [inline dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_fees.sql:1]: should this be made a view, if not we are bound to run into the same S3 write limits? :thinking:

Comment (0xBoxer) [reply dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_fees.sql:1]: oh I didn't catch that, will do

**Comment by 0xBoxer** [comment]
Due to a dbt version change, you need to merge main into your branch before the Github action runner runs successfully again.

**Comment by 0xBoxer** [comment]
Split monolithic Solana gas fees model into 5 files
- gas_solana.fees is now a view that combines regular and vote tx fees
- Separate models for decoding compute limit and unit price decoding
- vote tx fees are handled independently

File: dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_fees.sql
Comment (jeff-dude) [reply dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_fees.sql:1]: will be interesting to see how performant the view is to query

File: dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_tx_fees.sql [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        COALESCE(cl.compute_limit, 200000) AS compute_limit,
+        'So11111111111111111111111111111111111111112' AS tx_fee_currency,
+        b.leader
+    FROM {{ source('solana', 'transactions') }} t
+    LEFT JOIN {{ ref('solana_compute_limit') }} cl
+        ON t.id = cl.tx_id
+        AND t.block_date = cl.block_date
+    LEFT JOIN {{ ref('solana_compute_unit_price') }} up
+        ON t.id = up.tx_id
+        AND t.block_date = up.block_date
```
[the remark this answers is not in the document]
Comment (0xBoxer) [reply dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/solana/gas_solana_tx_fees.sql:31]: no you are right, an incremental condition on the join should help

**Comment by 0xRobin** [comment]
Tested some daily aggregations on the split models and the union view and performance is pretty good. 
Will scale up the cluster and give this another try in production :pray:

**Review by 0xRobin — APPROVED** [review]

**Pull request merged**